docs/specification/2025-03-26/basic/authorization.mdx

```diff
@@ -8,48 +8,141 @@ title: Authorization
 
 ### 1.1 Purpose and Scope
 
-The Model Context Protocol provides authorization capabilities at the transport level,
-enabling MCP clients to make requests to restricted MCP servers on behalf of resource
-owners. This specification defines the authorization flow for HTTP-based transports.
-
-### 1.2 Protocol Requirements
+The Model Context Protocol provides authorization capabilities at the transport level, enabling MCP
+clients to make requests to protected MCP servers on behalf of resource owners. This specification
+defines the authorization flow for HTTP-based transports.
+
+The scope of this document is **user authorization** - how a MCP client can authenticate a user
+to access a MCP server and the MCP server determine whether to authorize the user to access its tools,
+resources, or any other capabilities.
+
+This document **does not discuss server-to-server authorization patterns**.
+
+### 1.2 Baseline Tenets
+
+1. **MCP server developers are not expected to be security experts.**. Developers should not be
+   building their own authorization servers unless absolutely necessary, as they expect that MCP
+   provides a basline secure authentication and authorization process.
+1. **MCP clients are Authorization Server (AS)-agnostic**. When developers build MCP servers and secure
+   them with an AS, they need to design them so that standard, off-the-shelf MCP clients can connect
+   without needing any special modifications for the server's authorization setup.
+   Not doing this will fragment the ecosystem. The specification explicitly assumes that clients are
+   not enabled in any unique way to support specific AS implementations beyond what is covered here.
+    - **Note:** It is acceptable for clients to be enhanced for specific Authorization Servers,
+      primarily through pre-registration.
+1. **User Authentication is not a MCP Server concern.** The MCP server can guide the client
+   to the AS but will not be responsible for authenticating the user and obtaining credential
+   artifacts for the user.
+1. **We focus on OAuth 2.1, with room for extensibility for other providers/specifications.** Our
+   guiding assumption is that most developers will want to use OAuth 2.1, however in the long-run
+   unusual cases for authentication and authorization may emerge (e.g., mTLS, Kerberos), which will
+   need clients to be extensible.
+
+### 1.3 Protocol Requirements
 
 Authorization is **OPTIONAL** for MCP implementations. When supported:
 
-- Implementations using an HTTP-based transport **SHOULD** conform to this specification.
-- Implementations using an STDIO transport **SHOULD NOT** follow this specification, and
-  instead retrieve credentials from the environment.
-- Implementations using alternative transports **MUST** follow established security best
-  practices for their protocol.
+- Streamable HTTP or SSE transports **SHOULD** conform to this specification.
```

Comment:
this leaves out websockets, which would also want to handle this. The former wording was more permissive